glusterd2/config.go

```diff
@@ -18,39 +16,48 @@ import (
 	config "github.com/spf13/viper"
 )
 
+const (
+	defaultlogfile = "STDOUT"
+	defaultpeerport = "24008"
+	defaultpeeraddress = ":24008"
+	defaultclientaddress = ":24007"
+	defaultloglevel = "debug"
+)
+
 var (
 	// metrics
-	expConfig = expvar.NewMap("config")
-)
+	expConfig           = expvar.NewMap("config")
 
-const (
-	defaultLogLevel = "debug"
-	defaultConfName = "glusterd2"
+	// Following variables value set by LDFLAGS
+	defaultworkdir = ""
+	defaultlocalstatedir = ""
+	defaultlogdir = ""
+	defaultrundir = ""
 )
 
 // parseFlags sets up the flags and parses them, this needs to be called before any other operation
 func parseFlags() {
-	flag.String("workdir", "", "Working directory for GlusterD. (default: current directory)")
-	flag.String("localstatedir", "", "Directory to store local state information. (default: workdir)")
-	flag.String("rundir", "", "Directory to store runtime data.")
-	flag.String("config", "", "Configuration file for GlusterD. By default looks for glusterd2.toml in [/usr/local]/etc/glusterd2 and current working directory.")
+	flag.String("workdir", defaultworkdir, "Working directory for GlusterD.")
+	flag.String("localstatedir", defaultlocalstatedir, "Directory to store local state information.")
+	flag.String("rundir", defaultrundir, "Directory to store runtime data.")
+	flag.String("config", "", "Configuration file for GlusterD.")
 
-	flag.String(logging.DirFlag, "", logging.DirHelp+" (default: workdir/log)")
-	flag.String(logging.FileFlag, "STDOUT", logging.FileHelp)
-	flag.String(logging.LevelFlag, defaultLogLevel, logging.LevelHelp)
+	flag.String(logging.DirFlag, defaultlogdir, logging.DirHelp)
+	flag.String(logging.FileFlag, defaultlogfile, logging.FileHelp)
+	flag.String(logging.LevelFlag, defaultloglevel, logging.LevelHelp)
 
 	// TODO: Change default to false (disabled) in future.
 	flag.Bool("statedump", true, "Enable /statedump endpoint for metrics.")
 
-	flag.String("clientaddress", "", "Address to bind the REST service.")
-	flag.String("peeraddress", "", "Address to bind the inter glusterd2 RPC service.")
+	flag.String("clientaddress", defaultclientaddress, "Address to bind the REST service.")
+	flag.String("peeraddress", defaultpeeraddress, "Address to bind the inter glusterd2 RPC service.")
 
 	// TODO: SSL/TLS is currently only implemented for REST interface
 	flag.String("cert-file", "", "Certificate used for SSL/TLS connections from clients to glusterd2.")
 	flag.String("key-file", "", "Private key for the SSL/TLS certificate.")
 
 	// PID file
-	flag.String("pidfile", "", "PID file path(default: rundir/gluster/glusterd2.pid)")
+	flag.String("pidfile", "", "PID file path. (default \"rundir/glusterd2.pid)\"")
```

Comment:
```go
flag.String("pidfile", fmt.Sprintf("%s/glusterd2.pid", defaultrundir), "PID file path")
```

Reply:
thought about it, but that will be wrong when `--rundir` is specified